src/redis/driver_impl.go

```diff
@@ -73,6 +74,31 @@ func NewPoolImpl(scope stats.Scope, socketType string, url string, poolSize int)
 		stats: newPoolStats(scope)}
 }
 
+func NewAuthTLSPoolImpl(scope stats.Scope, auth string, url string, poolSize int) Pool {
+	logger.Warnf("connecting to redis on tls %s with pool size %d", url, poolSize)
+	df := func(network, addr string) (*redis.Client, error) {
+		conn, err := tls.Dial("tcp", addr, &tls.Config{})
+		if err != nil {
+			return nil, err
+		}
+		client, err := redis.NewClient(conn)
+
+		if err != nil {
+			return nil, err
+		}
+		if err = client.Cmd("AUTH", auth).Err; err != nil {
+			client.Close()
+			return nil, err
+		}
+		return client, nil
+	}
+	pool, err := pool.NewCustom("tcp", url, 10, df)
```

Comment:
The pool size setting is getting ignored here

Reply:
Resolved this